src/composables/use-ui-state.js

```diff
@@ -0,0 +1,18 @@
+import { readonly, ref } from '@nuxtjs/composition-api'
+
+const isFilterSidebarVisible = ref(false)
```

Comment:
Are global refs safe to use this way?

Reply:
This is the solution for using composable for global state: if the ref was inside the function, then it would be recreated in every component that uses it. This way, all the components using `isFilterSidebarVisible` are getting (and updating) the same value.
Here's a helpful article from VueSchool: https://vueschool.io/articles/vuejs-tutorials/state-management-with-composition-api/

Reply:
Gotcha, yeah I understand the difference here and the why, I just didn't realize it was _possible_. In React you can't have global refs because you can't call hooks outside of components. Point for Vue! 🙂